### PR TITLE
Add dependencies to targets for shared build

### PR DESCRIPTION
**Pull request by jandrej**

### CODE REVIEW

File: src/numerics/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+../src/numerics/CMakeLists.txt# Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
```
Comment (white238) [inline src/numerics/CMakeLists.txt]: copy and paste error.

Comment (samuelpmishLLNL) [inline src/numerics/CMakeLists.txt:1]: is this a typo?

Comment (jandrej) [reply src/numerics/CMakeLists.txt]: how did this happen?! :D

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
LGTM

**Review by joshessman-llnl — APPROVED** [review]

**Comment by samuelpmishLLNL** [comment]
it looks like we might be missing a few other dependencies, at a glance:

1. `integrators` should depend on `infrastructure` too
2. `infrastructure` should depend on `fmt`
3. `physics_operators` should depend on `physics_utilities`, `numerics` and `fmt`

**Comment by jamiebramwell** [comment]
LGTM

**Comment by jamiebramwell** [comment]
LGTM

**Pull request merged**